src/csharp/Microsoft.Spark.Worker/Command/SqlCommandExecutor.cs

```diff
@@ -131,15 +125,17 @@ internal class PicklingSqlCommandExecutor : SqlCommandExecutor
                     //   |null|
                     //   |  11|
                     //   +----+
-                    object[] inputRows = PythonSerDe.GetUnpickledObjects(readStream);
+                    object[] inputRows = PythonSerDe.GetUnpickledObjects(inputStream, messageLength);
 
                     for (int i = 0; i < inputRows.Length; ++i)
                     {
                         // Split id is not used for SQL UDFs, so 0 is passed.
                         outputRows.Add(commandRunner.Run(0, inputRows[i]));
                     }
 
-                    WriteOutput(outputStream, outputRows);
+                    // most probably input > output. If not, pickler increases the buffer
```

Comment:
To me, "most probably" is not true because it entirely depends on the UDFs. I would rewrite it to something like (and I think the code is clear enough without `sizeHint` variable):

```C#
// The initial (estimated) buffer size for pickling rows is set to the size of input pickled rows
// because the number of rows are the same for both input and output.
WriteOutput(outputStream, outputRows, messageLength)
```